src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/XmlDescriptor.php

```diff
@@ -340,6 +341,23 @@ private function getContainerDefinitionDocument(Definition $definition, $id = nu
             $serviceXML->setAttribute('factory-method', $definition->getFactoryMethod());
         }
 
+        if ($factory = $definition->getFactory()) {
+            $serviceXML->appendChild($factoryXML = $dom->createElement('factory'));
+
+            if (is_array($factory)) {
+                if ($factory[0] instanceof Reference) {
+                    $factoryXML->setAttribute('service', (string) $factory[0]);
+                } elseif ($factory[0] instanceof Definition) {
+                    // XXX What should we put here?
```

Comment:
E.g. replace [`new Reference` by `new Definition` below](#diff-32dc721f81bacb275e76f4e2759c72cdR113) and it will work (not with the same semantic of course)